robotbase/src/main/java/com/team2357/frc2022/subsystems/TurretSubsystem.java

```diff
@@ -0,0 +1,241 @@
+        boolean isMagnetDetected = false;
+
+        if (m_arduinoHallEffectSensor.isConnected()) {
+            isMagnetDetected = !m_arduinoHallEffectSensor
+                    .getDeviceFieldBoolean(Constants.ARDUINO.TURRET_HALL_SENSOR_NAME, "state");
+        }
+
+        if (isMagnetDetected) {
+            m_isZeroed = true;
+        }
+
+        return isMagnetDetected;
+    }
+
+    @Override
+    public void periodic() {
+        if (isClosedLoopEnabled()) {
+            closedLoopPeriodic();
+        }
+
+        // For tuning
+        SmartDashboard.putNumber("SetPoint", m_setPoint);
+        SmartDashboard.putNumber("Encoder Pos", getTurretRotations());
+        SmartDashboard.putNumber("Encoder Vel", m_turretMotor.getEncoder().getVelocity());
+        SmartDashboard.putNumber("Output", m_turretMotor.getAppliedOutput());
+    }
+
+    public boolean hasTarget() {
+        return m_currentTarget != null;
+    }
+
+    public boolean isTargetLocked() {
+        if (!isClosedLoopEnabled()) {
+            return false;
+        }
+        return atSetPoint();
+    }
+
+    @Override
+    public void setClosedLoopEnabled(boolean enabled) {
+        if (enabled) {
+            return;
+        }
+        disableClosedLoop();
+    }
+
+    public void enableClosedLoop(VisionTargetSupplier targetSupplier) {
+        m_targetSupplier = targetSupplier;
+        super.setClosedLoopEnabled(true);
+    }
+
+    public void disableClosedLoop() {
+        super.setClosedLoopEnabled(false);
+        m_targetSupplier = null;
+        m_turretMotor.set(0);
+    }
+
+    public void closedLoopPeriodic() {
+        m_currentTarget = m_targetSupplier.getAsVisionTarget();
+
+        if (m_isZeroed) {
+            if (m_currentTarget != null) {
+
+                double setPoint = calculateSetPoint(m_currentTarget);
+                setTurretPosition(setPoint);
+            } else {
+                lookForTarget();
+            }
+        }
+
+        // Occasionally reset heading to reduce error overtime
+        if (isOnZero()) {
+            resetHeading();
+        }
+    }
+
+    /**
+     * Calculate setpoint based on target x value
+     * Assumes limelight is mounted centered on the turret
+     * 
+     * @param target The vision target.
+     * @return desired setpoint
+     */
+    private double calculateSetPoint(VisionTarget target) {
+        double setPoint = (target.getX() * m_config.m_rotationsPerDegree) + getTurretRotations();
+
+        // Cases if target is out of reach
+        if (setPoint < m_config.m_turretRotationsClockwiseSoftLimit) {
+            setPoint = m_config.m_turretRotationsCounterClockwiseSoftLimit;
+        } else if (setPoint > m_config.m_turretRotationsCounterClockwiseSoftLimit) {
+            setPoint = m_config.m_turretRotationsClockwiseSoftLimit;
+        }
+
+        return setPoint;
+    }
+
+    // Looks for vision target by bouncing between soft limits
+    private void lookForTarget() {
```

Comment:
Let's remove this function altogether. This kind of seeking behavior will just waste time.